### DISCUSSION TITLE
Using istio and tekton-pipelines

### DISCUSSION THREAD

**Discussion by rkimmel-hv** [tektoncd/pipeline#4722] [Q&A]
Right now I have tekton-pipelines installed through the cncf tekton-pipelines chart here https://github.com/cdfoundation/tekton-helm-chart using v0.23.0 and triggers v0.13.0. Currently when my task runs spin up pods the istio-proxy image is replaced by the nop image and everything exits successfully. I am currently trying to change my pipeline version to v0.34.0 using the install process listed in this repo and triggers to v0.19.1. Now when I run a taskrun the istio-proxy container is no longer being replaced by the nop image and the pod hangs around for 45 mins or so before finally exiting. Can I configure something to get the previous behavior or is there some way to exit the istio-proxy container after my task run code completes? I'd prefer NOT to send the /quitquitquit message to istio-proxy if I don't need to.

Edit 1: More digging into this and I think the nop image should still be overriding the istio-proxy image in the task run pod but I'm not sure why this isn't happening. Doesn't look like the pipeline controller pod has any answers either

Edit 2: Comparing versions of taskrun.go from v0.23.0 to current v0.34.0 there seems to be a decent amount of logic in the stopSidecars method. In the old version any non-step containers were considered as sidecars but in the newer version it seems the sidecars must be listed on the taskrun to be considered. Specifically this change
```
// do not continue if the TaskSpec had no sidecars
	if tr.Status.TaskSpec != nil && len(tr.Status.TaskSpec.Sidecars) == 0 {
		return nil
	}
```
Now while I'd rather not specify istio-proxy sidecars for every single taskRun object I can do this If it will actually work.

Edit 3: This change was added in v0.30.0 with the following commit
https://github.com/tektoncd/pipeline/commit/2a70aedf09b0279a7e77a27e317ccb725ce94a9a

**Comment by dibyom** [comment]
https://github.com/tektoncd/pipeline/pull/5067 might fix this
